src/actions/config.js

```diff
@@ -1,13 +1,20 @@
 import yaml from "js-yaml";
 import { Map, List, fromJS } from "immutable";
-import { trimStart, flow } from "lodash";
+import { trimStart, flow, get } from "lodash";
 import { authenticateUser } from "Actions/auth";
 import * as publishModes from "Constants/publishModes";
 
 export const CONFIG_REQUEST = "CONFIG_REQUEST";
 export const CONFIG_SUCCESS = "CONFIG_SUCCESS";
 export const CONFIG_FAILURE = "CONFIG_FAILURE";
 export const CONFIG_MERGE = "CONFIG_MERGE";
+ 
+const getConfig = () => {
+  const validTypes = { 'text/yaml': 'yaml', 'application/x-yaml': 'yaml' };
+  const configLink = document.querySelector('link[rel="cms-config-url"]');
+  const isValidType = link => link && validTypes[link.type];
```

Comment:
Looks like we're going to crash if there's no matching link element by attempting to access property "type" of `null`.

Reply:
@erquhart doesn't `link && validTypes[link.type]` short circuit to false when `link` is `null`?

Reply:
🤦‍♂️🤦‍♂️🤦‍♂️ 

Disregard!